test_integration/test_plugin_static_policy.py

```diff
@@ -0,0 +1,115 @@
+sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+from test_integration import geopm_context
+import geopmpy.io
+from test_integration import geopm_test_launcher
+from test_integration import util
+import geopmpy.topo
+
+# todo this function results in test launcher
+# invocation, and is called 3 times (1 by class and
+# in 2 decorator tests).  may be useful to add once logic
+def geopmadmin():
+    settings = {}
+    for option in ["--config-default", "--config-override"]:
+        try:
+            proc = subprocess.Popen(shlex.split("geopmadmin {}".format(option)),
+                                    stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+            config_file = proc.stdout.readline()
+            stdout = io.StringIO()
+            stderr = io.StringIO()
+            geopm_test_launcher.allocation_node_test('dummy -- cat {}'.format(config_file),
+                                                     stdout, stderr)
+            settings.update(json.loads(stdout.getvalue()))
+        except subprocess.CalledProcessError:
+            # config_file may not exist and will cause allocation_node_test to fail
+            pass
+    return settings
+
+def geopmadminagent():
+    geopm_system_config = geopmadmin()
+    return geopm_system_config['GEOPM_AGENT']
+
+def geopmadminagentpolicy():
+    geopm_system_config = geopmadmin()
+    stdout = io.StringIO()
+    stderr = io.StringIO()
+    geopm_test_launcher.allocation_node_test('dummy -- cat {}'.format(geopm_system_config['GEOPM_POLICY']),
+                                             stdout, stderr)
+    return json.loads(stdout.getvalue())
+
+class TestIntegrationPluginStaticPolicy(unittest.TestCase):
+    """Test the static policy enforcement feature of the currently
+       configured RM plugin.
+
+    """
+    @classmethod
+    def setUpClass(cls):
+        cls._tmp_files = []
+        cls._geopmadminagent = geopmadminagent()
+        cls._geopmadminagentpolicy = geopmadminagentpolicy()
+
+    @unittest.skipUnless(geopmadminagent() in ['frequency_map', 'energy_efficient'],
+                         'Requires environment defaut/override to be configured to cap frequency.')
+    def test_frequency_cap_enforced(self):
+        if self._geopmadminagent == 'frequency_map':
+            test_freq = self._geopmadminagentpolicy['FREQ_MAX']
+        elif self._geopmadminagent == 'energy_efficient':
+            test_freq = self._geopmadminagentpolicy['FREQ_FIXED']
+        current_freq = geopm_test_launcher.geopmread("MSR::PERF_CTL:FREQ board 0")
+        self.assertEqual(test_freq, current_freq)
+
+    @unittest.skipUnless(geopmadminagent() in ['power_governor','power_balancer'],
+                         'Requires environment defaut/override to be configured to cap power.')
+    def test_power_cap_enforced(self):
+        test_power = self._geopmadminagentpolicy['POWER_PACKAGE_LIMIT_TOTAL']
+        # todo need test_launcher to execute topo on compute node
```

Comment:
Issue for this too please.

Reply:
added a decorator to the test class to require the test be ran in an allocation to resolve this (same as the static (non plugin) policy integration test)